api/src/pcapi/core/finance/api.py

```diff
@@ -1495,11 +1502,64 @@ def _mark_as_processed(pricings: sqla_orm.Query) -> None:
                     continue
 
                 total = pricings.with_entities(sqla.func.sum(models.Pricing.amount)).scalar()
-                if not total:
+
+                all_current_incidents = (
+                    models.FinanceIncident.query.join(models.FinanceIncident.booking_finance_incidents)
+                    .join(models.BookingFinanceIncident.finance_events)
+                    .join(models.FinanceEvent.pricings)
+                    .outerjoin(models.Pricing.cashflows)
+                    .filter(
+                        models.FinanceIncident.venueId.in_(venue_ids),
+                        models.FinanceIncident.status == models.IncidentStatus.VALIDATED,
+                        models.Cashflow.id.is_(None),  # Add this line to filter incidents with cashflows
+                    )
+                    .all()
+                )
+
+                if not total or (total > 0):
+                    last_cashflow_age = 0
+
+                    if total > 0:
+                        override_incident_debit_note = any(
+                            incident.forceDebitNote for incident in all_current_incidents
+                        )
+
+                        last_cashflow = (
+                            models.Cashflow.query.filter(
+                                models.Cashflow.reimbursementPointId == reimbursement_point_id,
+                                models.Cashflow.bankAccountId == bank_account_id,
+                                models.Cashflow.status == models.CashflowStatus.PENDING,
+                            )
+                            .order_by(models.Cashflow.creationDate.desc())
+                            .one_or_none()
+                        )
+                        if last_cashflow:
+                            last_cashflow_age = (datetime.datetime.utcnow() - last_cashflow.creationDate).days
+
+                        if (
+                            last_cashflow_age < conf.DEBIT_NOTE_AGE_THRESHOLD_FOR_CASHFLOW
+                            or not override_incident_debit_note
```

Comment:
[bug ?] Si on a indiqué vouloir forcer la génération d'une note de débit, on ne doit pas passer dans ce bloc. Je pense que la condition est incorrecte.

```suggestion
                            and not override_incident_debit_note
```